internal/pkg/cli/app_show.go

```diff
@@ -129,24 +132,45 @@ func (o *showAppOpts) description() (*describe.App, error) {
 	if err != nil {
 		return nil, fmt.Errorf("list jobs in application %s: %w", o.name, err)
 	}
-	//1st approach takes 2 seconds to display the table
-	for _, env := range envs {
-		deployedJobs, err := o.deployStore.ListDeployedJobs(o.name, env.Name)
-		if err != nil {
-			return nil, fmt.Errorf("list deployed jobs %s: %w", o.name, err)
-		}
-		for _, job := range deployedJobs {
-			workloadEnvs[job] = append(workloadEnvs[job], env.Name)
-		}
-		deployedSvcs, err := o.deployStore.ListDeployedServices(o.name, env.Name)
-		if err != nil {
-			return nil, fmt.Errorf("list deployed services %s: %w", o.name, err)
-		}
-		for _, svc := range deployedSvcs {
-			workloadEnvs[svc] = append(workloadEnvs[svc], env.Name)
-		}
-	}
+	var mux sync.Mutex
+	g, ctx := errgroup.WithContext(context.Background())
+	defer ctx.Done()
+	for i := range envs {
+		env := envs[i]
+		g.Go(func() error {
```

Comment:
What do y'all think of a format like this?
```go
g.Go(func() error { 
  return o.populateDeployedWorkloads(o.deployStore.ListDeployedJobs, deployedEnvsFor, env.Name) 
})
g.Go(func() error { 
   return o.populateDeployedWorkloads(o.deployStore.ListDeployedServices, deployedEnvsFor, env.Name) 
})

func (o *showAppOpts) populateDeployedWorkloads(
   listWorkloads func(app, env string) ([]string, error),  
   deployedEnvsFor map[string][]string, 
   env string) error {
  // ...
}

```

Reply:
I like that a lot! Just don't forget the mutex :)